src/Rule/Nested.php

```diff
@@ -153,6 +177,7 @@ private function prepareRules(iterable|object|string|null $source): void
         $rules = $rules instanceof Traversable ? iterator_to_array($rules) : $rules;
         self::ensureArrayHasRules($rules);
 
+        /** @var iterable<RuleInterface> $rules */
```

Comment:
```suggestion
```

Use `@psalm-assert RuleInterface[] $rules` for `ensureArrayHasRules()` method.

Reply:
OK, but after deleting this https://github.com/yiisoft/validator/pull/364#discussion_r1019537918 it's no longer an array, but iterable.